cmake/common.cmake

```diff
@@ -134,50 +146,48 @@ macro(spl_create_component)
+        set(_sphinx_source_dir ${PROJECT_SOURCE_DIR})
+
 
         # Create component docs target if there is an index.rst file in the component's doc directory
         if(EXISTS ${_component_doc_file})
-            # The Sphinx source directory is always the project root
-            set(SPHINX_SOURCE_DIR ${PROJECT_SOURCE_DIR})
-            set(SPHINX_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/docs)
-            file(RELATIVE_PATH _rel_sphinx_output_dir ${SPHINX_SOURCE_DIR} ${SPHINX_OUTPUT_DIR})
-            set(SPHINX_OUTPUT_HTML_DIR ${SPHINX_OUTPUT_DIR}/html)
-            set(SPHINX_OUTPUT_INDEX_HTML ${SPHINX_OUTPUT_HTML_DIR}/index.html)
+            file(RELATIVE_PATH _rel_component_docs_out_dir ${_sphinx_source_dir} ${_component_docs_out_dir})
+            string(JSON _component_info SET "${_component_info}" docs_output_dir "\"${_rel_component_docs_out_dir}\"")
+            string(JSON _component_info SET "${_component_info}" has_docs "\"True\"")
+            set(_component_docs_html_out_dir ${_component_docs_out_dir}/html)
 
             # create the config.json file. This is exported as SPHINX_BUILD_CONFIGURATION_FILE env variable
-            set(_docs_config_json ${SPHINX_OUTPUT_DIR}/config.json)
-            file(RELATIVE_PATH _rel_component_doc_dir ${SPHINX_SOURCE_DIR} ${_component_doc_dir})
+            set(_docs_config_json ${_component_docs_out_dir}/config.json)
+            file(RELATIVE_PATH _rel_component_doc_dir ${_sphinx_source_dir} ${_component_doc_dir})
             file(WRITE ${_docs_config_json} "{
-                \"generated_rst_content\": \".. toctree::\\n    :maxdepth: 2\\n\\n    /${_rel_component_doc_dir}/index\",
-                \"component_doc_dir\": \"${_rel_component_doc_dir}\",
-                \"include_patterns\": [\"${_rel_component_doc_dir}/**\",\"${_rel_sphinx_output_dir}/**\"]
+                \"component_info\": ${_component_info},
+                \"include_patterns\": [\"${_rel_component_doc_dir}/**\",\"${_rel_component_docs_out_dir}/**\"]
             }")
 
             # add the generated files as dependency to cmake configure step
             set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${_docs_config_json})
             add_custom_target(
                 ${component_name}_docs
-                COMMAND ${CMAKE_COMMAND} -E make_directory ${SPHINX_OUTPUT_DIR}
-                COMMAND ${CMAKE_COMMAND} -E env SPHINX_BUILD_CONFIGURATION_FILE=${_docs_config_json} AUTOCONF_JSON_FILE=${_autoconf_json_file} VARIANT=${VARIANT} -- sphinx-build -b html ${SPHINX_SOURCE_DIR} ${SPHINX_OUTPUT_HTML_DIR}
-                BYPRODUCTS ${SPHINX_OUTPUT_INDEX_HTML}
+                COMMAND ${CMAKE_COMMAND} -E make_directory ${_component_docs_out_dir}
+                COMMAND ${CMAKE_COMMAND} -E env SPHINX_BUILD_CONFIGURATION_FILE=${_docs_config_json} AUTOCONF_JSON_FILE=${AUTOCONF_JSON} VARIANT=${VARIANT} -- sphinx-build -E -b html ${_sphinx_source_dir} ${_component_docs_html_out_dir}

```

Comment:
We do not know all dependencies for generating the docs (apart from the rst files). This might cause incremental builds to not update parts of the documentation. To avoid this we are using the -E option to make sphinx-build writing all files new.